src/Languages/Editor/Impl/ContainedLanguage/ContainedLanguageHost.cs

```diff
@@ -0,0 +1,44 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See LICENSE in the project root for license information
+
+using Microsoft.Languages.Editor.Composition;
+using Microsoft.Languages.Editor.Services;
+using Microsoft.VisualStudio.Text;
+using Microsoft.VisualStudio.Text.Editor;
+
+namespace Microsoft.Languages.Editor.ContainedLanguage {
+    /// <summary>
+    /// Helper class that simplifies location of contained language host
+    /// for contained (secondary) languages. 
+    /// </summary>
+    public static class ContainedLanguageHost {
```

Comment:
I.e. host is created for a given language. It has more functionality in general case like, say, C#. For example, it may supply editor settings from top-level language down so if Tabs are set in the primary, secondary should also be using Tabs in formatting. It may also supply base indent for formatting if needed so embedded language indents relatively to the primary language construct and not to just its buffer content. Fortunately, markdown doesn't have formatting so this is not needed. Yet.